### DISCUSSION TITLE
Bonvision: Switch between different rendering settings

### DISCUSSION THREAD

**Discussion by LeonKremers** [bonsai-rx/bonsai#1229] [Q&A] [answered]
Hi everyone,
I am creating a Bonsai script for displaying visual stimuli using Bonvision. This script contains multiple stimulus types wich can be selected from a menu and are displayed as soon as a master trigger input is detected on an DAQ. Most stimuli are 2D (e.g. moving bars, movies, full-field chirp) and are rendered using the "Orthographic View" node. However, we would like to also incude a simple VR corridor which we render using the "Render Cubemap" node. I tried to switch rendering settings like this: 
![Workflow](https://user-images.githubusercontent.com/70718367/212648172-f27d8068-fc98-4d1c-bd19-9a75f58c3690.png)

When I run the script and select the VR corridor it works for a few seconds before Bonsai freezes and gives me the following error message:
![Error](https://user-images.githubusercontent.com/70718367/212648456-a7fa14d9-f726-49c3-8388-1de66b74c59c.png)

I tried to add an "OnErrorResumeNext" node after the workflow but it still freezes, this time with a different error message:
![error 2](https://user-images.githubusercontent.com/70718367/212648757-e9cc4078-7f3b-4eed-b5d8-d10ce2fb0652.png)

Is there a way to get rid of these errors and display differently rendered stimuli with the same script?
(Here is my whole workflow: [bench2p_1_headfixed_combined_1.5.zip](https://github.com/bonsai-rx/bonsai/files/10424591/bench2p_1_headfixed_combined_1.5.zip) )

Thanks,
Leon

**Comment by bruno-f-cruz** [comment, marked as answer]
I can't test the workflow since I dont have the hardware you are using. Can you try to render the problem down to the simplest workflow that reproduces the bug?

Meanwhile, I see alot of Timer(rx) instead of Timer(Shader) in your workflow. When using Timer try to always use the one from the shaders when working with the shaders packages. Same for other nodes that use that scheduler (e.g. Delay, DelaySubscription, Throttle, etc...)

**Reply by LeonKremers** [reply]
Hi Bruno, thank you for your reply. I replaced every timer with Timer(Shader). Bonsai still crashed but I think I found the error while minimizing the workflow for you. I used PublishSubject to broadcast the input from the rotary encoder to the VR corridor workflow and also some other parts of the script. When I inserted the encoder input directly in the VR corridor workflow it works without freezing after a while. This makes constantly logging the rotary encoder a bit more difficult since the VR corridor workflow is mostly inactive, but I will figure something out.

Thank you very much for your help!
